app/src/common/components/article-list/title.jsx

```diff
@@ -2,32 +2,46 @@ import {connect} from 'react-redux';
 import {searchArticleList} from '../../actions/article-list';
 import i18n from 'i18next';
 import {TextField, LinearProgress} from 'material-ui';
+import {PropTypes} from 'react';
 
-export default connect(
-    state => ({
-        filter: state.articleList.filter,
-        loading: state.articleList.isLoading,
-        error: state.articleList.error
-    }),
-    dispatch => ({search: filter => dispatch(searchArticleList(filter))})
-)(function ArticleListTitle({loading, filter, search, error, textFieldUnderline}) {
+const defaultProps = {
+    textFieldUnderline: true
+};
+
+const propTypes = {
+    textFieldUnderline: PropTypes.bool
+};
+
+const ArticleListTitle = ({loading, filter, search, error, textFieldUnderline}) => {
     return (
         <div className='article-list-header'>
             <h3>{i18n.t('article-list.title')}</h3>
             <div className='search'>
                 <i className='material-icons'>search</i>
                 <div className='search-bar'>
                     <TextField
-                        underlineShow={textFieldUnderline !== undefined ? textFieldUnderline : !(loading && filter)}
+                        underlineShow={textFieldUnderline && !(loading && filter)}
                         errorText={error ? ' ' : null}
                         errorStyle={{color: 'indianred'}}
                         value={filter}
                         onChange={e => search(e.target.value)}
                         hintText={i18n.t('search.placeholder')}
                     />
-                    {textFieldUnderline ? (loading && filter ? <LinearProgress style={{position: 'absolute', bottom: '8px', height: '2px'}} /> : null) : null}
+                    {textFieldUnderline && loading && filter ? <LinearProgress style={{position: 'absolute', bottom: '8px', height: '2px'}} /> : null}
                 </div>
             </div>
         </div>
     );
-});
+};
+
+ArticleListTitle.defaultProps = defaultProps;
+ArticleListTitle.propTypes = propTypes;
+
+export default connect(
+    state => ({
+        filter: state.articleList.filter,
+        loading: state.articleList.isLoading,
+        error: state.articleList.error
+    }),
+    dispatch => ({search: filter => dispatch(searchArticleList(filter))})
+)(ArticleListTitle);
```

Comment:
Use proper functions instead of lambdas.
And I remember we did something short to handle the case where `textFieldUnderline` was undefined instead of having to write these ugly as hell `defaultProps`.